### ISSUE TITLE
CA1853: Should not report if `Dictionary.ContainsKey()` is required

### ISSUE THREAD

**Issue by SetTrend** [dotnet/roslyn-analyzers#7490]
# Analyzer

**Diagnostic ID**: &nbsp; [CA1853](https://learn.microsoft.com/en-us/dotnet/fundamentals/code-analysis/quality-rules/ca1853) **"Unnecessary call to `Dictionary.ContainsKey(key)`"**

# Current Situation

In my [`Dictionary`](https://learn.microsoft.com/en-us/dotnet/api/system.collections.generic.dictionary-2) object, a particular flagging item is added to the collection as a return value item for a subsequent method to execute a special piece of code. The sole existence of this item is used to recognize when to execute a piece of code differently. After the special treatment has occured, I want to remove the item from the collection, so all subsequent computations will continue as usual.

#### Here's the piece of code:

```c#
case "div":
  if (attributes.ContainsKey("Floater"))
  {
    attributes.Remove("Floater");
    xmlWriter.WriteStartElement("Floater");
  }
  else xmlWriter.WriteStartElement("Section");
  break;
```

This excerpt currently triggers an analyzer hint: "Unnecessary call to `Dictionary.ContainsKey(key)`". But that's not true. The call to [`Dictionary.ContainsKey(key)`](https://learn.microsoft.com/en-us/dotnet/api/system.collections.generic.dictionary-2.containskey) is necessary as there is other code depending on the existence of the item.

# Desired Situation

The analyzer should not trigger a hint when [`Dictionary.Remove(key)`](https://learn.microsoft.com/en-us/dotnet/api/system.collections.generic.dictionary-2.remove) is not the only action encapsulated by the `if`-branch.
